### PR TITLE
NAS-109138 / 21.02 / fix up gluster.volume/peer and ctdb.shared.volume plugins

### PR DESCRIPTION
**Pull request by yocalebo**
Since the gluster and ctdb related APIs aren't stable, yet. I've gone ahead and done a few things:

- make gluster.volume/peer plugins follow our standard for CRUD apis
- add a new plugin that is simply a wrapper for calling external library for gluster related information
- make ctdb.shared.volume and gluster.volume/peer plugins use new wrapper
- convert ctdb.shared.volume and gluster.volume/peer to `async`
- removed unnecessary methods in gluster.volume/peer plugins respectively
- updated doc strings to be uniform between the plugins
- along the way I've fixed a few various small bugs

### CODE REVIEW

File: src/middlewared/middlewared/plugins/gluster_linux/peer.py
Comment (william-gr) [inline src/middlewared/middlewared/plugins/gluster_linux/peer.py:20]: From your CLI changes, we should not use =None here, right @themylogin ?

File: src/middlewared/middlewared/plugins/gluster_linux/peer.py
Comment (william-gr) [inline src/middlewared/middlewared/plugins/gluster_linux/peer.py:113]: Why would it not be bytes?

File: src/middlewared/middlewared/plugins/gluster_linux/peer.py
Comment (william-gr) [inline src/middlewared/middlewared/plugins/gluster_linux/peer.py:87]: We seem to be doing xml things here like parsing and finding. I wonder if it would make sense to keep it in a method and not a coroutine so the event loop is not potentially blocked?

File: src/middlewared/middlewared/plugins/gluster_linux/run.py
Comment (william-gr) [inline src/middlewared/middlewared/plugins/gluster_linux/run.py:32]: We probably dont need f-string here

File: src/middlewared/middlewared/plugins/gluster_linux/peer.py
Comment (themylogin) [reply src/middlewared/middlewared/plugins/gluster_linux/peer.py:20]: Yes, let's get rid of this, thank you, @william-gr

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/gluster_linux/peer.py]: Do I need to wait until the CLI changes are merged? @themylogin @william-gr

Comment (themylogin) [reply src/middlewared/middlewared/plugins/gluster_linux/peer.py]: No, you can merge it now

**Review by william-gr — APPROVED** [review]

**Pull request merged**
